File: VP Semester  Project/Project Attempts/IPTest1/IPTest1/MainActivity.cs
using System;
using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
using Android.Net.Wifi;
using Android.Net;
using Android.Net.Rtp;
using System.Net;
using System.Net.NetworkInformation;
using System.Net.Sockets;
using System.Threading;
using Java.Net;
using Android.Media;
using Android.Util;

namespace IPTest1
{
    [Activity(Label = "IPTest1", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity
    {
        EditText info,info2;
        WifiManager wifii;
        DhcpInfo d;
        string ip;
       
        string mac;
        AudioStream audi;
       
        InetAddress iin;
        InetAddress ini;
        AudioGroup audiogroup;
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            // Get our button from the layout resource,
            // and attach an event to it

            info = FindViewById<EditText>(Resource.Id.editText1);
            wifii = (WifiManager)GetSystemService(Context.WifiService);
            d = wifii.DhcpInfo;
            ip = Convert.ToString(IntToIp(d.IpAddress));
            iin = InetAddress.GetByName(ip);

           
            
            string[] array = ip.Split('.');
            for (int i = 0; i <=255; i++)
            {
                string Broadcast = array[0] + "." + array[1] + "." + array[2] + "." + i;
                ini = InetAddress.GetByName(Broadcast);
                Thread childThread = new Thread(() => send(iin, ini));
                childThread.Start();
            }
            
            
            
           
            WifiInfo wa = wifii.ConnectionInfo;           
            info2 = FindViewById<EditText>(Resource.Id.editText2);
            //mac = wa.MacAddress + "\n BSSID " + wa.BSSID + "\n Link Speed " + wa.LinkSpeed + "\n Network Id " + wa.NetworkId + "\n" + IntToIp(wa.IpAddress);
            //info2.Text = mac;
            //info.Text =" BroadCast Address"+ Broadcast.ToString();
            info.Text = ini.ToString();
           // ThreadStart childref = new ThreadStart(send(iin,ini));
            
        }
        public string IntToIp(int ip)
        {
            return
                (ip & 0xFF) + "." +
                ((ip >> 8) & 0xFF) + "." +
                ((ip >> 16) & 0xFF) + "." +
                ((ip >> 24) & 0xFF);                
                   
                 
        }
        public void send(InetAddress iin,InetAddress ini)
        {
            try
            {
                AudioManager audio = (AudioManager)GetSystemService(Context.AudioService);
                audio.SpeakerphoneOn = true;
                audio.Mode = Mode.InCommunication;
                audiogroup = new AudioGroup();
                audiogroup.Mode = AudioGroupMode.Normal;
                audi = new AudioStream(iin);   //To Be Continued
                audi.Mode = RtpStreamMode.Normal;
                audi.Associate(ini, audi.LocalPort);
                audi.Codec = AudioCodec.Pcmu;
                audi.Join(audiogroup);
            }
            catch (Java.Net.SocketException s) { s.PrintStackTrace(); }
            catch(UnknownHostException uk) { uk.PrintStackTrace(); }
            catch(Exception e) { Log.Error(" Error at :", e.StackTrace); }
        }
       
        
    }
}

File: VP Semester  Project/Project Attempts/RtpReceiver/app/src/main/java/com/example/muhammadwaleed/rtpreceiver/MainActivity.java
package com.example.muhammadwaleed.rtpreceiver;

import android.content.Context;
import android.media.AudioManager;
import android.net.rtp.AudioCodec;
import android.net.rtp.AudioGroup;
import android.net.rtp.AudioStream;
import android.net.rtp.RtpStream;
import android.os.StrictMode;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import java.net.InetAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.util.Enumeration;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
        StrictMode.setThreadPolicy(policy);
        try {
            AudioManager audio =  (AudioManager) getSystemService(Context.AUDIO_SERVICE);
            audio.setMode(AudioManager.MODE_IN_COMMUNICATION);
            AudioGroup audioGroup = new AudioGroup();
            audioGroup.setMode(AudioGroup.MODE_NORMAL);
            AudioStream audioStream = new AudioStream(InetAddress.getByAddress(getLocalIPAddress()));
            audioStream.setCodec(AudioCodec.PCMU);
            audioStream.setMode(RtpStream.MODE_NORMAL);
            audioStream.associate(InetAddress.getByAddress(new byte[] {(byte)192, (byte)168, (byte)1, (byte)19 }), 22222);
            audioStream.join(audioGroup);


        } catch (Exception e) {
            Log.e("----------------------", e.toString());
            e.printStackTrace();
        }
    }
    public static byte[] getLocalIPAddress () {
        byte ip[]=null;
        try {
            for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements();) {
                NetworkInterface intf = en.nextElement();
                for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements();) {
                    InetAddress inetAddress = enumIpAddr.nextElement();
                    if (!inetAddress.isLoopbackAddress()) {
                        ip= inetAddress.getAddress();
                    }
                }
            }
        } catch (SocketException ex) {
            Log.i("SocketException ", ex.toString());
        }
        return ip;

    }

}
